repo2docker/utils.py

```diff
@@ -469,6 +469,10 @@ def is_local_pip_requirement(line):
     if line.startswith(("--requirement", "--constraint")):
         # as above but flags are spelt out
         return True
+    # the `--pre` flag is a global flag and should appear on a line by itself
+    # we just care that this isn't a "local pip requirement"
+    if line.startswith("--pre"):
+        return False
     # strip off things like `--editable=`. Long form arguments require a =
     if line.startswith("--"):
         line = line.split("=", 1)[1]
```

Comment:
both here and L479 we have splits without first checking if the split character (`=` here, any space using `None` on L479) is present, meaning that `split(X, 1)[1]` can raise an IndexError. We should protect any such split from raising, either with:

- try/except or
- checking for the character's presence before splitting (harder with space), or
- checking len after split before requesting `item[1]`

In general: we shouldn't assume requirements.txt is valid and such errors should be raised by pip on `pip install`, not with errors in our own code.

Reply:
Is this what you had in mind? I didn't wrap the whole function in a try/except because I thought that might swallow errors we might actually want to see, but unsure.

I also went with returning `True` in the error case because either it is something we don't understand so better safe than sorry or it will later turn out to be a syntax error in which case the return value doesn't matter. WDYT?

Reply:
I'm inclined to return False because if there's no argument being passed, I think it's very unlikely that it's expressing a local dependency and more likely to be something configuring pip that isn't relevant to us.